js/io.ts

```diff
@@ -159,3 +161,31 @@ export function toAsyncIterator(r: Reader): AsyncIterableIterator<Uint8Array> {
     }
   };
 }
+
+/** Read lines asynchronously from `r`.
+ *
+ *      for await (const line of Deno.lines(reader)) {
+ *          console.log(line);
+ *      }
+ */
+export async function* lines(r: Reader): AsyncIterableIterator<string> {
+  const decoder = new TextDecoder();
+  let scratch = "";
+
+  for await (const chunk of toAsyncIterator(r)) {
+    scratch += decoder.decode(chunk);
+
+    let newLineIndex = scratch.indexOf("\n");
+
+    if (newLineIndex > -1) {
+      const line = scratch.slice(0, newLineIndex);
+      scratch = scratch.slice(newLineIndex + 1);
+      yield line;
```

Comment:
doesn't only one yield per chunk mean scratch could grow linearly?
I think this newLineIndex/yield should be in a while loop.